src/main.rs

```diff
@@ -1,183 +1,153 @@
 #![feature(panic_info_message)]
 
-#[macro_use] extern crate failure;
 #[macro_use] extern crate serde_derive;
 
-mod format;
 mod lookup;
 mod models;
 mod utils;
 
-use self::models::*;
+use models::ExtendSelectionParams;
```

Comment:
This should be moved to `selectionRanges` API: https://github.com/gluon-lang/lsp-types/blob/bc714cd8836c8bb0f910cd32e87b98c09c721944/src/request.rs#L610-L618